### DISCUSSION TITLE
Rotate and Mirror Live Video

### DISCUSSION THREAD

**Discussion by Dailura** [raspberrypi/picamera2#976] [Q&A] [answered]
Hello, 
I am a total beginner in Python language. I am using the "examples/mjpeg_server.py" project to stream my video on a webserver.

My problem is, that the video is upside down (because my camera also is upside down). But for my project the camera has to stay like that. 

My question now is: 
**How can I flip the video vertically and horizontally?**

I hope you guys can help me =)

**Comment by CTBradshawJr** [comment, marked as answer]
I am new to Python as well, but have found the following works:

Using mjpeg_server.py, Insert the following before line 16
  from libcamera import Transform

replace original line 87: picam2.configure(picam2…. With the following 2 lines:
  config = picam2.create_video_configuration(main={"size":(640,480)}, transform=Transform(hflip = True, vflip = True))
  picam2.configure(config)

**Reply by Dailura** [reply]
Thanks alot bro =) 
It's working

**Reply by ArosPrince** [reply]
Can you flip in runtime, or do you always have to stop and start the camera for that? All the examples I've been able to find set this beforehand. Some other params like `ScalerCrop` can be set during runtime, but `transform` doesn't seem to be one of those... 

Thanks for any help on this one.
